memory/src/main/scala/filodb.memory/BlockMemFactoryPool.scala

```diff
@@ -15,32 +17,54 @@ class BlockMemFactoryPool(blockStore: BlockManager,
                           metadataAllocSize: Int,
                           baseTags: Map[String, String]) extends StrictLogging {
   private val factoryPool = new collection.mutable.Queue[BlockMemFactory]()
+  private val checkedOut = new mutable.HashMap[Int, BlockMemFactory]()
 
   def poolSize: Int = factoryPool.length
```

Comment:
Should make this synchronized too while you're at it.

Reply:
👍